de.metas.report/report-service/src/main/java/de/metas/adempiere/report/jasper/JasperEngine.java

```diff
@@ -93,17 +102,21 @@
 	private static final String PARAM_OUTPUTTYPE = "OUTPUTTYPE";
 
 	// services
+	private final UserAuthTokenRepository userAuthTokenRepo = Adempiere.getBean(UserAuthTokenRepository.class);
+	private final JsonDataSourceService jsonDSService = Adempiere.getBean(JsonDataSourceService.class);
+	private final JsonDataSourceRepository jsonRepo = Adempiere.getBean(JsonDataSourceRepository.class);
```

Comment:
avoid using the repository direct.
You sould call only the service and the service shall call the repo